introduction_and_learning/Histogram/histograms.py

```diff
@@ -0,0 +1,291 @@
+__all__ = [
+    "HistogramTxt",
+    "HistogramJSON",
+    "HistogramXml"
+]
+
+from histogram_api import HistogramInterface
+import json
+import xml.etree.ElementTree as ET
+import csv
+
+
+class HistogramTxt(HistogramInterface):
+    def read_data(self, file_path: str) -> str:
+        extension = file_path[file_path.rfind("."):]
+
+        if extension == ".txt":
+            with open(file_path, "tr", encoding="utf8") as f:
+                return f.read()
+        else:
+            return "File does not have a .txt extension."
```

Comment:
I would rather raise some custom crafted error here instead of returning some string.
As User wants to use `HistogramTxt.read_data()` on non `.txt` file, we should clearly tell him/her that this is illegal. Do you agree?
Anyway, we should use exceptions in situations when it is something wrong, this will be more readable and better exposed to the user.